### PR TITLE
added a back button to return to the list of MCs

### PR DESCRIPTION
**Pull request by eddiebroad**
I also adjusted some whitespace so things don't go over 100 characters/columns
in a single row

### CODE REVIEW

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 1]
```diff
@@ -104,7 +104,12 @@
                      "Destination Namespace"
                      (style/create-text-field {:defaultValue selected-conf-namespace :ref "destinationNamespace"})}]
           (create-formatted-label-textfield k v))
-        (render-import-button props refs)]))})
+        (render-import-button props refs)
+        [:span {:style {:marginLeft "0.5em"}}
+         [comps/Button
+          {:text "Back"
+           :onClick (fn [e]
+                      (apply (:on-back props)nil))}]]]))})
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: Just do `:onClick #((:on-back props))`.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 2]
```diff
@@ -149,24 +154,34 @@
...
+                                                       :show-import-mc-modal?
+                                                       :selected-method-config)
+                                                     (swap! state assoc
+                                                       :loaded-import-confs? true))
+                                          :on-import (fn [& args]
+                                                       (swap! state dissoc
+                                                         :show-import-mc-modal?)
+                                                       (apply (:on-import props) args))}]
+        (:loaded-import-confs? @state) [ConfigurationsTable {:method-configs
+                                                                (:method-configs @state)
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: To save line length, bring down ConfigurationImportForm and ConfigurationsTable to the next line.  Then you can merge some of these lines back together.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 3]
```diff
@@ -149,24 +154,34 @@
...
+        (:show-import-mc-modal? @state) [ConfigurationImportForm
+                                         {:selected-method-config (:selected-method-config @state)
+                                          :workspace-id (:workspace-id props)
+                                          :on-back (fn []
+                                                     (swap! state dissoc
+                                                       :show-import-mc-modal?
+                                                       :selected-method-config)
+                                                     (swap! state assoc
+                                                       :loaded-import-confs? true))
+                                          :on-import (fn [& args]
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: I don't know why this function is taking a variable number of arguments.  It's always a function of one argument: the imported config.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 4]
```diff
@@ -149,24 +154,34 @@
...
-                                                                                     :loaded-import-confs? false))}]
+                                                                (swap! state assoc
+                                                                  :selected-method-config config
+                                                                  :show-import-mc-modal? true
+                                                                  :loaded-import-confs? false))}]
         (:error-message @state) (style/create-server-error-message (:error-message @state))
         :else [comps/Spinner {:text "Loading configurations for import..."}])])
-   :component-did-mount
-   (fn [{:keys [state]}]
+   :component-did-mount (fn [{:keys [state]}]
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: Revert this.  Our pattern is for the function to be on the next line.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs (hunk 3 above)
Comment (eddiebroad) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: I don't know why either.  But I am left everything alone here except for whitespace.  Would such a refactoring be outside the scope of this PR?

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 5]
```diff
@@ -175,10 +190,14 @@
   (react/create-element
     [:div {}
      [:div {:style {:position "absolute" :right 2 :top 2}}
-      [:div {:style {:backgroundColor (:button-blue style/colors) :color "#fff" :padding "0.5em" :cursor "pointer"}
+      [:div {:style {:backgroundColor
+                     (:button-blue style/colors)
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: Don't split the key and value like this.  These style properties easily fit on 2 lines.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs (hunk 3 above)
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: I don't think so.  I think all that needs to happen is to change this declaration to `(fn [config] ...)`, and then change where it's called to `((:on-import props) config)`.

Comment (eddiebroad) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: I wonder would you be interested to take over this PR and  branch ?

**Comment by eddiebroad** [comment]
@MattPutnam  I wonder would you be interested to take over this PR and branch ?

**Comment by MattPutnam** [comment]
I don't think that's necessary.  This functionality works.  I don't like the layout of the buttons but that can be taken care of later.

**Pull request closed without merging**